pkg/kv/kvserver/client_raft_test.go

```diff
@@ -1606,26 +1606,23 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
 			}
 		}
 		// Determine which node to propose on. Transfer lease to that node.
-		var propIdx int
-		if !proposeOnFollower {
-			propIdx = 0
-		} else {
+		propIdx := 0
+		if proposeOnFollower {
 			propIdx = 1
 		}
 		propNode := tc.GetFirstStoreFromServer(t, propIdx).TestSender()
 		tc.TransferRangeLeaseOrFatal(t, *leaderRepl.Desc(), tc.Target(propIdx))
+		tc.WaitForLeaseUpgrade(ctx, t, *leaderRepl.Desc())
 		testutils.SucceedsSoon(t, func() error {
 			// Lease transfers may not be immediately observed by the new
 			// leaseholder. Wait until the new leaseholder is aware.
 			repl, err := tc.GetFirstStoreFromServer(t, propIdx).GetReplica(leaderRepl.RangeID)
-			if err != nil {
-				t.Fatal(err)
-			}
+			require.NoError(t, err)
 			repDesc, err := repl.GetReplicaDescriptor()
-			if err != nil {
-				t.Fatal(err)
-			}
-			if lease, _ := repl.GetLease(); !lease.Replica.Equal(repDesc) {
+			require.NoError(t, err)
+			lease, _ := repl.GetLease()
+			t.Logf("lease: %v", lease)
```

Comment:
Assuming this was just for debugging, remove?

Reply:
Done.